assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Overview.test.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * Overview component tests.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+	render,
+	createTestRegistry,
+	provideModules,
+	provideUserCapabilities,
+	provideUserInfo,
+	provideUserAuthentication,
+	provideSiteInfo,
+	waitFor,
+} from '../../../../../../../tests/js/test-utils';
+import coreModulesFixture from '../../../../../googlesitekit/modules/datastore/__fixtures__';
+import { CORE_MODULES } from '../../../../../googlesitekit/modules/datastore/constants';
+import { CORE_USER } from '../../../../../googlesitekit/datastore/user/constants';
+import { MODULES_SEARCH_CONSOLE } from '../../../datastore/constants';
+import { getWidgetComponentProps } from '../../../../../googlesitekit/widgets/util';
+import Overview from './Overview';
+
+jest.mock( '../../../../../hooks/useBreakpoint', () => ( {
+	BREAKPOINT_SMALL: 'small',
+	useBreakpoint: () => 'tablet',
+} ) );
+
+describe( 'Overview', () => {
+	let registry;
+
+	const { WidgetReportError } = getWidgetComponentProps( 'searchFunnel' );
+
+	const searchConsoleData = [
+		{
+			clicks: 123,
+			ctr: 4.56,
+			impressions: 7890,
+			keys: [ '2022-06-21' ],
+			position: 12.345,
+		},
+	];
+
+	const overviewProps = {
+		searchConsoleData,
+		selectedStats: 0,
+		handleStatsSelection: () => {},
+		dateRangeLength: 28,
+		WidgetReportError,
+	};
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+
+		provideModules( registry );
+		provideUserInfo( registry );
+		provideUserAuthentication( registry );
+		provideUserCapabilities( registry );
+		provideSiteInfo( registry );
+		registry.dispatch( CORE_USER ).receiveConnectURL( 'test-url' );
+		registry.dispatch( MODULES_SEARCH_CONSOLE ).receiveGetSettings( {
+			propertyID: 'http://example.com/',
+		} );
+
+		fetchMock.getOnce(
+			/^\/google-site-kit\/v1\/modules\/search-console\/data\/searchanalytics/,
+			{
+				body: searchConsoleData,
+			}
+		);
+	} );
+
+	it( 'should render the Search Funnel Overview, including the Activate Analytics CTA', async () => {
+		const { container, getByText } = await waitFor( () =>
```

Comment:
If the `waitFor` is here due to registry state changes, you can wait for those using `waitForRegistry` returned by our `render` instead 👍

Reply:
Thanks for pointing this out, I'll change to `waitForRegistry`.

Incidentally I had copied this `waitFor` usage from another test, so I have created a new issue to update the existing tests - https://github.com/google/site-kit-wp/issues/5734. Please take a look, feel free to close if you think it's unnecessary!